Add option to set multi_db on TestCase

morty opened the pull request:
By default fixtures are only loaded into the `default` database. If you have multiple databases in order to get the fixtures loaded into all of the databases you have to set `multi_db=True` on the TestCase.

This change allows you to specify multi_db in the `before_scenario` function.

bittner commented:
Please add a test and update the documentation. Thanks!

mixxorz commented:
I think now's a good time to explore adding an API to modify the test instance before it's set up.

bittner commented:
@morty Would you care to add a test and update the documentation? A directly link to the [related Django documentation](https://docs.djangoproject.com/en/2.0/topics/testing/tools/#multi-database-support) would be nice. Please, rebase and squash the commits for a cleaner history, if possible.

bittner commented on features/steps/fixture-loading.py:
```diff
@@ -30,3 +30,8 @@ def check_second_fixtures(context):
+
+
+@then(u'multi_db should be enabled')
+def check_reset_seqeunces(context):
```
This function name has a typo.

bittner commented on features/steps/fixture-loading.py:
```diff
@@ -30,3 +30,8 @@ def check_second_fixtures(context):
 def check_reset_seqeunces(context):
```
Can you also fix this typo in the function name, while you're at it? Thank you!

morty commented:
@bittner thanks for the feedback. I have rebased and squashed as requested.

The tests are failing now. I think that is due to changes in behave 1.2.6 - even master fails now. With behave 1.2.6 installed it complains about not being able to find steps.

```
ConfigError: No steps directory in '<directory where tests are run from>'
```

bittner commented on features/steps/fixture-loading.py:
```diff
@@ -27,6 +27,11 @@ def check_second_fixtures(context):
 
 @then(u'the sequences should be reset')
-def check_reset_seqeunces(context):
+def check_reset_sequences(context):
```
Good catch, thanks!

bittner commented on features/steps/fixture-loading.py:
```diff
@@ -27,6 +27,11 @@ def check_second_fixtures(context):
+
+@then(u'multi_db should be enabled')
+def check_multi_db(context):
+    assert context.test.multi_db
```
I'm actually a friend of plain `assert`, but I fear we need to use boilder-plate `assertEquals` to stay consistent with the rest of the code.  :disappointed:

morty replied:
Very good point. I have updated this now.

bittner commented on features/fixture-loading.feature:
```diff
@@ -13,3 +13,6 @@ Feature: Fixture loading
         Then the sequences should be reset
+
+    Scenario: Load fixtures with multi_db option
+        Then multi_db should be enabled
```
Maybe I'm just tired, but ... shouldn't there also be a `When` to set the `multi_db` option in the test?

morty replied:
It can't be done in a `When`. If it could then this change would not be needed. It has to be done in `before_scenario` to be early enough.

bittner commented:
@morty Re the `ConfigError`, weird: If the change comes from [lines 749+ in behave](https://github.com/behave/behave/blob/master/behave/runner.py#L749-L750) then this requirement is 2 or even 4 years old. Hmmm...

But we definitely need to do something about it, so that the build can be proven to pass. Either pin an old version of *behave* form now, or extend the build matrix.

bittner commented:
@mixxorz Can you look at this PR, too, please?

bittner commented:
We've just merged #62. Also, the project layout has changed (that's why your PR has conflicts).

Would you want to rebase this PR, bring it up to date with the new structure and verify how your changes fit together with what came in from #62? That would be great!

bittner commented:
@morty Specifically note https://github.com/behave/behave-django/pull/62#issuecomment-390127329, please.

bittner commented:
@morty Any interest to take this PR further? Or should we close it?

morty commented:
@bittner I don't have access to the repo I made the PR from. I'm just in the process of getting access again.

I've rebased the 1 original commit on my own repo without pinning the version of behave in the tox config:

https://github.com/morty/behave-django/tree/multi-db-option

It seems to apply OK and the tests still pass for me.

I'm not sure if you can change the from branch of this PR or if a new PR would be easier or if I should get access to the repo I originally made the PR from?

bittner commented:
It probably easier to close this PR in favor of a new PR.

morty commented:
Closed see https://github.com/behave/behave-django/pull/79
